packages/transcript-view/src/transcript-view.ts

```diff
@@ -0,0 +1,383 @@
+    `;
+  }
+
+  private get bottomContextZoneDevTemplate(): TemplateResult {
+    return html`
+      <div class="bottom context-overlay" style="height: ${this.bottomContextHeight}px"></div>
+    `;
+  }
+
+  get currentEntryStartTime(): number {
+    return this.currentEntry ? this.currentEntry.start : 0;
+  }
+
+  static get styles(): CSSResult {
+    const transcriptHeightCss = css`var(--transcriptHeight, 200px)`;
+
+    const timeColorCss = css`var(--timeColor, white)`;
+    const timeColumnWidthCss = css`var(--timeColumnWidth, 5rem)`;
+
+    const autoScrollButtonFontColorCss = css`var(--autoScrollButtonFontColor, black)`;
+    const autoScrollButtonBackgroundColorCss = css`var(--autoScrollButtonBackgroundColor, white)`;
+
+    return css`
+      .container {
+        position: relative;
+      }
+
+      .auto-scroll-button {
+        position: absolute;
+        left: 0;
+        right: 0;
+        margin: auto;
+        width: 8rem;
+        bottom: 1rem;
+        border-radius: 1rem;
+        border: 0;
+        color: ${autoScrollButtonFontColorCss};
+        background-color: ${autoScrollButtonBackgroundColorCss};
+      }
+
+      .context-overlay {
+        position: absolute;
+        left: 0;
+        width: 100%;
+        height: 0;
+        z-index: -1;
+        display: block;
+      }
+
+      .context-overlay.top {
+        top: 0;
+        border-bottom: 1px solid green;
+      }
+
+      .context-overlay.bottom {
+        bottom: 0;
+        border-top: 1px solid green;
+      }
+
+      .time {
+        flex: 0 0 ${timeColumnWidthCss};
+        color: ${timeColorCss};
+        position: relative;
+      }
+
+      .time-display {
+        position: absolute;
+        top: 0;
+      }
+
+      .scroll-container {
+        display: flex;
+        overflow-y: auto;
+        font-size: 16px;
+        line-height: 24px;
+        -ms-overflow-style: none;
+        scrollbar-width: none;
+        height: ${transcriptHeightCss};
+      }
+
+      .scroll-container::-webkit-scrollbar {
+        display: none;
+      }
+    `;
+  }
+
+  private transcriptEntrySelected(e: CustomEvent): void {
+    const { entry } = e.detail;
+    const event = new CustomEvent('transcriptEntrySelected', {
+      detail: { entry },
+      bubbles: true,
+      composed: true,
+    });
+    this.dispatchEvent(event);
+    this.selectedSearchResultIndex = entry.searchMatchIndex;
+    this.autoScroll = false;
+  }
+
+  private handleCurrentTimeChange(): void {
+    const entries = this.entries ? this.entries : [];
+    const activeEntry = entries.find(
+      // eslint-disable-next-line max-len
+      (entry: TranscriptEntryConfig) =>
+        this.currentTime >= entry.start && this.currentTime <= entry.end,
+    );
+
+    if (!activeEntry) {
+      this.currentEntry = undefined;
+      return;
+    }
+
+    if (this.currentEntry && this.currentEntry.id === activeEntry.id) {
```

Comment:
I'm a bit confused about the usage of this conditional. if we have matching IDs between `this.currentEntry` and `activeEntry`, then we return, but the only statement left in this block is an assignment of the active entry to the current entry anyway. Can we just remove this entirely?

Reply:
We don't want to change the `currentEntry` if it's the same one that we're already on. This method gets called quite often (many times per second) but we don't need it to update until we reach the next entry. I added a comment about this.